src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -147,18 +151,31 @@ export class EditorPart extends Part implements IEditorPart {
 		@IThemeService themeService: IThemeService,
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IStorageService storageService: IStorageService,
-		@IWorkbenchLayoutService layoutService: IWorkbenchLayoutService
+		@IWorkbenchLayoutService layoutService: IWorkbenchLayoutService,
 	) {
 		super(id, { hasTitle: false }, themeService, storageService, layoutService);
 
 		this.registerListeners();
 	}
 
 	private registerListeners(): void {
+		this._register(this.onDidSetGridWidget(() => {
+			this._register(this.onDidChangeActiveGroup(() => this.showActiveEditorGroupsIfHidden()));
+		}));
 		this._register(this.configurationService.onDidChangeConfiguration(e => this.onConfigurationUpdated(e)));
 		this._register(this.themeService.onDidFileIconThemeChange(() => this.handleChangedPartOptions()));
 	}
 
+	// If hidden group becomes active, show all groups
+	private showActiveEditorGroupsIfHidden(): void {
+		for (const group of this.groups) {
```

Comment:
I wonder if we should move this code closer to where a group becomes active? Here:

```
// Expand the group if it is currently minimized
			this.doRestoreGroup(group);
```